stixcore/io/fits/processors.py

```diff
@@ -111,52 +136,23 @@ def generate_primary_header(cls, filename, product):
         tuple
             List of header cards as tuples (name, value, comment)
         """
-        if product.level != 'LB':
-            raise ValueError(f"Try to crate FITS file LB for {product.level} data product")
-        if 'scet_coarse' not in product.control:
-            raise ValueError("Expected scet_coarse in the control structure")
-
-        obs_beg = SCETime(coarse=product.control["scet_coarse"][0],
-                          fine=product.control["scet_fine"][0]).as_float().value
-        obs_end = SCETime(coarse=product.control["scet_coarse"][-1],
-                          fine=product.control["scet_fine"][-1]).as_float().value
+        # if product.level != 'LB':
+        #     raise ValueError(f"Try to crate FITS file LB for {product.level} data product")
+        # if 'scet_coarse' not in product.control:
+        #     raise ValueError("Expected scet_coarse in the control structure")
 
         headers = FitsProcessor.generate_common_header(filename, product) + (
             # Name, Value, Comment
-            ('OBT_BEG', str(obs_beg), 'Start of acquisition time in OBT'),
-            ('OBT_END', str(obs_end), 'End of acquisition time in OBT'),
+            ('OBT_BEG', product.obt_beg.to_string(), 'Start of acquisition time in OBT'),
+            ('OBT_END', product.obt_end.to_string(), 'End of acquisition time in OBT'),
             ('TIMESYS', 'OBT', 'System used for time keywords'),
             ('LEVEL', 'LB', 'Processing level of the data'),
-            ('DATE_OBS', str(obs_beg), 'Start of acquisition time in OBT'),
-            ('DATE_BEG', str(obs_beg), 'Start of acquisition time in OBT'),
-            ('DATE_END', str(obs_end), 'End of acquisition time in OBT')
+            ('DATE_OBS', product.obt_beg.to_string(), 'Start of acquisition time in OBT'),
+            ('DATE_BEG', product.obt_beg.to_string(), 'Start of acquisition time in OBT'),
+            ('DATE_END', product.obt_end.to_string(), 'End of acquisition time in OBT')
         )
```

Comment:
Much cleaner now. Did i forget to change it here?

Reply:
No I think it just slightly different now because the str and to_string are different because of the `:` in the file name issue which is gone now alway. As part of a clean up could just make the `__str__` method work as it was.